perun-core/src/main/java/cz/metacentrum/perun/core/implApi/SearcherImplApi.java

```diff
@@ -41,6 +41,27 @@
 	 */
 	List<User> getUsers(PerunSession sess, Map<Attribute, String> attributesWithSearchingValues) throws InternalErrorException;
 
+	/**
+	 * This method get Map of Attributes with searching values and try to find all groups, which have specific attributes in format.
+	 * Better information about format below. When there are more than 1 attribute in Map, it means all must be true "looking for all of them" (AND)
+	 *
+	 * IMPORTANT: can't get CORE ATTRIBUTES
+	 *
+	 * @param sess perun session
+	 * @param attributesWithSearchingValues map of attributes
+	 *        when attribute is type String, so value is string and we are looking for total match (Partial is not supported now, will be supported later by symbol *)
+	 *        when attribute is type Integer, so value is integer in String and we are looking for total match
+	 *        when attribute is type List<String>, so value is String and we are looking for at least one total or partial matching element
+	 *        when attribute is type Map<String> so value is String in format "key=value" and we are looking total match of both or if is it "key" so we are looking for total match of key
+	 *        IMPORTANT: In map there is not allowed char '=' in key. First char '=' is delimiter in MAP item key=value!!!
+	 * @return list of groups who have attributes with specific values (behavior above)
+	 *        if no group exist, return empty list of groups
+	 *        if attributeWithSearchingValues is empty, return all groups
+	 *
+	 * @throws InternalErrorException
+	 */
+	List<Group> getGroups(PerunSession sess, Map<Attribute, String> attributesWithSearchingValues) throws InternalErrorException;
```

Comment:
``InternalErrorException`` is runtime. We are removing all ocurrences in other pull-request.

Reply:
You are right. Removed.